modules/machineset-troubleshooting-azure-ultra-disk.adoc

```diff
@@ -0,0 +1,96 @@
+:pvc:
+endif::[]
+ifeval::["{context}" == "persistent-storage-csi-azure"]
+:pvc:
+endif::[]
+
+:_content-type: REFERENCE
+[id="machineset-troubleshooting-azure-ultra-disk_{context}"]
+= Troubleshooting resources for machine sets that enable ultra disks
+
+Use the information in this section to understand and recover from issues you might encounter.
+
+ifdef::pvc[]
+[id="ts-pvc-mounting-ultra_{context}"]
+== Unable to mount a persistent volume claim backed by an ultra disk
+
+If there is an issue mounting a persistent volume claim backed by an ultra disk, the pod becomes stuck in the `ContainerCreating` state and an alert is triggered.
+
+For example, if the `UltraSSDEnabled` parameter is not set on the machine that backs the node that hosts the pod, the following error message appears:
+
+[source,terminal]
+----
+StorageAccountType UltraSSD_LRS can be used only when additionalCapabilities.ultraSSDEnabled is set.
+----
+
+* To resolve this issue, describe the pod by running the following command:
++
+[source,terminal]
+----
+$ oc ~something~
+----
+#How does the user "describe the pod" to resolve this?#
+endif::pvc[]
+
+ifdef::mapi[]
+[id="ts-mapi-attach-misconfigure_{context}"]
+== Incorrect ultra disk configuration
+
+If an incorrect configuration of the `ultraSSDEnabled` parameter is specified in the machine set, the machine provisioning fails.
+
+For example, if the `ultraSSDEnabled` parameter is set to `Disabled`, but an ultra disk is specified in the `dataDisks` parameter, the following error message appears:
+
+[source,terminal]
+----
+StorageAccountType UltraSSD_LRS can be used only when additionalCapabilities.ultraSSDEnabled is set.
+----
+
+* To resolve this issue, verify that your machine set configuration is correct.
+
+[id="ts-mapi-attach-unsupported_{context}"]
+== Unsupported disk parameters
+
+If a region, availability zone, or instance size that is not compatible with ultra disks is specified in the machine set, the machine provisioning fails. Check the logs for the following error message:
+#Is this for if the user specifies an incompatible parameter, if they specify a compatible one that is not actually what they have, or both?#
+
+[source,terminal]
+----
+vm size xyz does not support ultra disks in location xyz. select a different vm size or disable ultra disks
+----
+#This message did not seem to be exact wording in the enhancement, is exact wording available?#
+
+* To resolve this issue, verify that you are using this feature in a supported environment and that your machine set configuration is correct.
+
+[id="ts-mapi-delete_{context}"]
+== Unable to delete disks
+
+If the deletion of ultra disks as data disks is not working as expected, the machines are deleted and the data disks are orphaned. Check the logs for the following error message:
+
+[source,terminal]
+----
+failed to delete Data Disk: xyz
+----
+#This message did not seem to be exact wording in the enhancement, is exact wording available?#
```

Comment:
Need some info here
Cc @damdo

Reply:
I'll need to get back to you on this.

Reply:
This changed slightly when we added the `DeletionPolicy` option.
The Data Disk deletion is now managed by the Azure Cloud magic rather than issued by the MAPI Azure controller. So I'd keep the:
```
If the deletion of ultra disks as data disks is not working as expected, the machines are deleted and the data disks are orphaned
```
just as an informative thing on what could happen, but I'd delete the:
```
Check the logs for the following error message:

[source,terminal]
----
failed to delete Data Disk: xyz
```
bits.